### PR TITLE
Do not disable warnings by default when compiling tests

### PR DESCRIPTION
**Pull request by shindere**
The main thing this PR does is stop disabling the compiler warnings while
compiling the programs in the testsuite.

Only one test program really did produce a warning so the code has been
updated to make this warning go away.

Is this okay with you @mshinwell ?

Who else is concerned by this test?

### CODE REVIEW

**Comment by shindere** [comment]
Working on an updated version, no need to review yet. Will comment when ready.

**Comment by shindere** [comment]
Rebased on latest trunk and massively updated, but still not ready.

See commit log for the details.

How should the warnings in testsuite/tests/basic-more/morematch.ml
be dealt with?

A certain number of warnings are declared as "normal" at the beginnng of
the file, but one gets more. Should all warnings be considered as
expected, or not?

@pierre-weiss @xavierleroy @damiendoligez ?

**Comment by damiendoligez** [comment]
All the warnings are noted as normal in the file (in randomly interspersed comments) except:
* line 231
* line 444
* line 469

I think you should treat all the warnings as expected in this file.
@pierreweis (correct spelling)

**Comment by pierre-weiss** [comment]
Dear Damien,

just an email to tell you that you picked up the wrong Pierre Weis. I'm
Pierre Weiss, mathematician in Toulouse.

Cheers,
Pierre


2017-08-28 16:58 GMT+02:00 Damien Doligez <notifications@github.com>:

**Comment by damiendoligez** [comment]
FTR: I got the right Pierre, and Sebastien has the wrong one...

**Comment by shindere** [comment]
PR rebased and updated to take into account @damiendoligez 's comment.

Should be ready for review and merge now.

**Comment by shindere** [comment]
Damien Doligez (2017/08/28 14:58 +0000):
> All the warnings are noted as normal in the file (in randomly interspersed comments) except:
> * line 231
> * line 444
> * line 469
> 
> I think you should treat all the warnings as expected in this file.

Done.

> @pierreweis (correct spelling)

Sorry about that, thanks for correcting!

File: testsuite/tests/basic-more/tformat.ml [hunk 1]
```diff
@@ -1,21 +0,0 @@
-(*
-
```
Comment (damiendoligez) [inline testsuite/tests/basic-more/tformat.ml:2]: Why did you remove this test?

**Review by damiendoligez — APPROVED** [review]
Can you tell us why you removed a test? Other than that, LGTM.

**Comment by shindere** [comment]
Damien Doligez (2017/08/30 16:04 +0000):
> damiendoligez approved this pull request.

Thanks.

> Can you tell us why you removed a test?

Sure. It's because the function under test is deprecated, as documented
in the commit log.

**Review by maranget — APPROVED** [review]
On behalf of Damien

**Comment by shindere** [comment]
Damien Doligez (2017/08/31 12:24 +0000):
> Merged #1293.

Thanks! Didn't dare to merge because of Travis' failure.
